src/index.ts

```diff
@@ -38,10 +38,7 @@ export async function main(argv: string[]) {
   );
 
   for (const choice of response.choices) {
-    if (
-      choice.finish_reason === "function_call" &&
-      choice.message.function_call
-    ) {
+    if (choice.message.function_call) {
```

Comment:
when would the finish reason be something else when providing function? 🤔

Reply:
I know it's not clear in the documentation, I figured this out when I implemented a function-based chat bot with function calling, but it does respond with the `stop` reason if it doesn't expect to need the result of that function.

And for the shell function it never needs the result, every time it responds with the `function_call` stop reason it's an error, because it has no instruction to do something with the output of the command.

Try this input: "show which processes have connections to which hosts", it consistently gets the "stop" finish_reason.

![image](https://github.com/searchableguy/whiz/assets/59587503/6ff61c9c-9e4c-4d47-9bf7-baf5e3504bfc)

Reply:
Thanks. That make sense.